Gcpe.Hub.API/Controllers/SocialMediaController.cs

```diff
@@ -0,0 +1,147 @@
+    {
+        private readonly HubDbContext dbContext;
+        private readonly ILogger<SocialMediaController> logger;
+        private readonly IMapper mapper;
+
+        public SocialMediaController(HubDbContext dbContext, ILogger<SocialMediaController> logger, IMapper mapper)
+        {
+            this.dbContext = dbContext;
+            this.logger = logger;
+            this.mapper = mapper;
+        }
+
+        [HttpGet]
+        [Produces(typeof(SocialMediaPostViewModel[]))]
+        [ProducesResponseType(400)]
+        public IActionResult GetAll()
+        {
+            try
+            {
+                var posts = dbContext.SocialMediaPost.Where(p => p.IsActive).OrderBy(p => p.SortOrder).ToList();
+                return Ok(mapper.Map<List<SocialMediaPost>, List<SocialMediaPostViewModel>>(posts));
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Failed to get social media posts: {ex}");
+                return BadRequest("Failed to get social media posts");
+            }
+        }
+
+        [HttpPost]
+        [Produces(typeof(SocialMediaPostViewModel))]
+        [ProducesResponseType(201)]
+        [ProducesResponseType(400)]
+        public IActionResult Post(SocialMediaPostViewModel postVM)
+        {
+            try
+            {
+                if (postVM.Id != Guid.Empty)
+                {
+                    throw new ValidationException("Invalid parameter (id)");
+                }
+                SocialMediaPost post = mapper.Map<SocialMediaPostViewModel, SocialMediaPost>(postVM);
+                post.Id = Guid.NewGuid();
+                post.IsActive = true;
+                dbContext.SocialMediaPost.Add(post);
+                dbContext.SaveChanges();
+                return CreatedAtRoute("GetPost", new { id = post.Id }, mapper.Map<SocialMediaPost, SocialMediaPostViewModel>(post));
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Failed to create social media post: {ex}");
+                return BadRequest("Failed to create social media post");
+            }
+        }
+
+        [HttpGet("{id}", Name = "GetPost")]
+        [Produces(typeof(SocialMediaPostViewModel))]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        public IActionResult Get(Guid id)
+        {
+            try
+            {
+                var post = dbContext.SocialMediaPost.Find(id);
+                if (post != null && post.IsActive)
+                {
+                    return Ok(mapper.Map<SocialMediaPost, SocialMediaPostViewModel>(post));
+                }
+                return NotFound($"Social media post not found with id: {id}");
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Failed to retrieve social media post: {ex}");
+                return BadRequest("Failed to retrieve social media post");

```

Comment:
Maybe put this string as constant but replace retrieve with {operation} and use it everywhere

Reply:
Can you show me how to do this in C#?